Transcom/Parsers/Interfaces/IParser.cs

```diff
@@ -0,0 +1,9 @@
+﻿using System.Collections.Generic;
+
+namespace Transcom.Parsers
+{
+    public interface IParser<T> where T : class
+    {
+        List<T> ParseInput(string[] input, string[] conditionalCollection);
```

Comment:
`conditionalCollection` is not a descriptive name of the parameter. Try using a name that describes what info is stored in that collection. If there is no suitable name that encompasses all use cases, maybe a name like `additionalArguments` can be more appropriate

Reply:
Actually, what is the purpose of the IParser<T> interface? Do you need it?